File: C++ DEVELOPMENT/C++ Basics/Regular-Exam/05.Everest.cpp
#include <iostream>
#include <string>
#include <algorithm> // For transform
using namespace std;

int main() {
    string command;
    int daysCounter = 1;
    int metersCounter = 5364;
    bool isSucceeded = false;

    while (true) {
        cin >> command;
        if (command == "END") {
            break;
        }

        string isSpendingTheNight = command;
        transform(isSpendingTheNight.begin(), isSpendingTheNight.end(), isSpendingTheNight.begin(), ::tolower);
        if (isSpendingTheNight == "yes") {
            daysCounter++;
        }

        int climbedMeters;
        cin >> climbedMeters;
        if (daysCounter > 5) {
            isSucceeded = false;
            break;
        }
        metersCounter += climbedMeters;

        if (metersCounter >= 8848) {
            isSucceeded = true;
            break;
        }
    }

    if (isSucceeded) {
        cout << "Goal reached for " << daysCounter << " days!" << endl;
    }
    else {
        cout << "Failed!" << endl;
        cout << metersCounter << endl;
    }

    return 0;
}

File: CSharp Development/C# Introduction - Jan 2026/16. Exam/05.Everest.cs
using System;

int days = 1;
int currentHeight = 5364;
const int EverestHeight = 8848;

string command = Console.ReadLine()!;

while (command != "END")
{
    if (command == "Yes")
    {
        days++;

        if (days > 5)
        {
            break;
        }
    }

    int climbedMeters = int.Parse(Console.ReadLine()!);
    currentHeight += climbedMeters;

    if (currentHeight >= EverestHeight)
    {
        break;
    }

    command = Console.ReadLine()!;
}

if (currentHeight >= EverestHeight)
{
    Console.WriteLine($"Goal reached for {days} days!");
}
else
{
    Console.WriteLine("Failed!");
    Console.WriteLine(currentHeight);
}
